Nvim-dap crashes on launch

The problem seems to be nvim-dap is confused about receiving a reverse-request for "runInTerminal" before receiving a response for "launch"

there are 3 types of messages: requests/responses/events

every request must have a response. events just happen

however, there are a few "special requests" because they go from the server to the client (the server requests something of the client). In this case, it requests to run something in the terminal.

But nvim-dap crashes with `[ERROR] 2026-05-12 10:41:58 dap/session.lua:1090    "No callback found. Did the debug adapter send duplicate responses?"` before we even send the launch response

This could be fixed in nvim-dap or in hdb. though we'd have to understand at what point does nvim-dap start allowing reverse-requests to do a workaround.

This likely regressed in the runInTerminal refactor, which made the runInTerminal reverse request get sent before se reply with the launch response. It may be easy to get back to sending the launch response eagerly even while the program is still loading.

Here's the server log and nvim-dap log:
```
$ hdb server --port 12345
Running DAP server on 12345...
TCP connection established from 127.0.0.1:60267
[127.0.0.1:60267][DEBUG][RECEIVED]
{
    "arguments": {
        "adapterID": "nvim-dap",
        "clientID": "neovim",
        "clientName": "neovim",
        "columnsStartAt1": true,
        "linesStartAt1": true,
        "locale": "en_US.UTF-8",
        "pathFormat": "path",
        "supportsProgressReporting": true,
        "supportsRunInTerminalRequest": true,
        "supportsStartDebuggingRequest": true,
        "supportsVariableType": true
    },
    "command": "initialize",
    "seq": 1,
    "type": "request"
}
[127.0.0.1:60267][DEBUG][SENT]
{
    "body": {
        "additionalModuleColumns": [
            {
                "attributeName": "Extra",
                "label": "Label"
            }
        ],
        "completionTriggerCharacters": [],
        "exceptionBreakpointFilters": [
            {
                "filter": "break-on-exception",
                "label": "All exceptions"
            },
            {
                "filter": "break-on-error",
                "label": "Uncaught exceptions"
            }
        ],
        "supportSuspendDebuggee": false,
        "supportTerminateDebuggee": false,
        "supportedChecksumAlgorithms": [],
        "supportsBreakpointLocationsRequest": true,
        "supportsCancelRequest": false,
        "supportsClipboardContext": false,
        "supportsCompletionsRequest": false,
        "supportsConditionalBreakpoints": true,
        "supportsConfigurationDoneRequest": true,
        "supportsDataBreakpoints": false,
        "supportsDelayedStackTraceLoading": false,
        "supportsDisassembleRequest": false,
        "supportsEvaluateForHovers": false,
        "supportsExceptionFilterOptions": false,
        "supportsExceptionInfoRequest": true,
        "supportsExceptionOptions": true,
        "supportsFunctionBreakpoints": true,
        "supportsGotoTargetsRequest": false,
        "supportsHitConditionalBreakpoints": true,
        "supportsInstructionBreakpoints": false,
        "supportsLoadedSourcesRequest": false,
        "supportsLogPoints": true,
        "supportsModulesRequest": false,
        "supportsReadMemoryRequest": false,
        "supportsRestartFrame": false,
        "supportsRestartRequest": false,
        "supportsSetExpression": false,
        "supportsSetVariable": false,
        "supportsSingleThreadExecutionRequests": false,
        "supportsStepBack": false,
        "supportsStepInTargetsRequest": false,
        "supportsSteppingGranularity": false,
        "supportsTerminateRequest": true,
        "supportsTerminateThreadsRequest": false,
        "supportsValueFormattingOptions": true,
        "supportsWriteMemoryRequest": false
    },
    "command": "initialize",
    "request_seq": 1,
    "seq": 2,
    "success": true,
    "type": "response"
}
[127.0.0.1:60267][DEBUG][RECEIVED]
{
    "arguments": {
        "debugServer": 12345,
        "entryArgs": [],
        "entryFile": "Main.hs",
        "entryPoint": "main",
        "extraGhcArgs": [],
        "name": "Haskell Debugger",
        "projectRoot": "/private/var/folders/_j/l1f15qm555z4qzvjjg6ftp_c0000gn/T/tmp.Mhob87dla7",
        "request": "launch",
        "type": "haskell-debugger"
    },
    "command": "launch",
    "seq": 2,
    "type": "request"
}
[2026-05-12T10:41:58.123670Z] [INFO] LogSetupMsg "Figuring out the right flags to compile the project using hie-bios..."
[2026-05-12T10:41:58.127569Z] [INFO] LogCradle Cradle{ cradleRootDir = "/private/var/folders/_j/l1f15qm555z4qzvjjg6ftp_c0000gn/T/tmp.Mhob87dla7", cradleOptsProg = CradleAction: Default}
[2026-05-12T10:41:58.127658Z] [INFO] LogSetupMsg "Checking GHC version against debugger version..."
...
[2026-05-12T10:41:58.223034Z] [INFO] LogSetupMsg "Session setup with hie-bios was successful."
[127.0.0.1:60267][DEBUG][SENT]
{
    "body": {
        "category": "console",
        "output": "libdir: /nix/store/zxnh5wx0w6n6rlj44r3kakkv3rx5fr3v-ghc-9.14.1/lib/ghc-9.14.1/lib\nunits: \nargs: -dynamic -fno-it\n"
    },
    "event": "output",
    "type": "event"
}
[127.0.0.1:60267][DEBUG][SENT]
{
    "body": {
Trying to connect to 60268
        "output": "Figuring out the right flags to compile the project using hie-bios...\n"
    },
    "event": "output",
    "type": "event"
}
[127.0.0.1:60267][INFO][Registered new debug session: debug-session:b7e659c5-9405-4c7f-b661-9c3bc08eab62]
[127.0.0.1:60267][DEBUG][SENT]
{
    "body": {
        "output": "Using cradle: Cradle{ cradleRootDir = \"/private/var/folders/_j/l1f15qm555z4qzvjjg6ftp_c0000gn/T/tmp.Mhob87dla7\", cradleOptsProg = CradleAction: Default}\n"
    },
    "event": "output",
    "type": "event"
}
[127.0.0.1:60267][DEBUG][SENT]
{
    "arguments": {
        "args": [
            "/Users/romes/.local/state/cabal/store/ghc-9.14.1-7f12/hskll-dbggr-0.13.1.0-50f13547/bin/hdb",
            "external-interpreter",
            "--port",
            "60268"
        ],
        "argsCanBeInterpretedByShell": false,
        "cwd": "",
        "kind": "integrated"
    },
    "command": "runInTerminal",
    "request_seq": 2,
    "seq": 3,
    "type": "request"
}
[127.0.0.1:60267][DEBUG][SENT]
{
    "body": {
        "output": "Checking GHC version against debugger version...\n"
    },
    "event": "output",
    "type": "event"
}
[127.0.0.1:60267][DEBUG][SENT]
{
    "command": "launch",
    "request_seq": 2,
    "seq": 3,
    "success": true,
    "type": "response"
}
[127.0.0.1:60267][DEBUG][SENT]
{
    "body": {
        "output": "Discovering session flags with hie-bios...\n"
    },
    "event": "output",
    "type": "event"
}
[127.0.0.1:60267][DEBUG][SENT]
{
    "event": "initialized",
    "type": "event"
}
[127.0.0.1:60267][DEBUG][SENT]
{
    "body": {
        "output": "invoking build tool to determine build flags (this may take some time depending on the cache)\n"
    },
    "event": "output",
    "type": "event"
}
[2026-05-12T10:41:58.226333Z] [INFO] Debugger launched successfully.
[127.0.0.1:60267][DEBUG][SENT]
{
    "body": {
        "output": "Session setup with hie-bios was successful.\n"
    },
    "event": "output",
    "type": "event"
}
[127.0.0.1:60267][INFO][SENT][Client has ended its connection]
[127.0.0.1:60267][INFO][SENT][Closing Connection]
```

Nvim:
```
[ERROR] 2026-05-12 10:41:58 dap/session.lua:1090	"No callback found. Did the debug adapter send duplicate responses?"	{
  command = "launch",
  request_seq = 2,
  seq = 3,
  success = true,
  type = "response"
}
```

--- a/haskell-debugger.cabal
+++ b/haskell-debugger.cabal
@@ -197,7 +197,7 @@ library dap
         network-run >= 0.4.4,
         async >= 2.2.5 && < 2.3,
         text >= 2.1 && < 2.3,
-        dap >= 0.6 && < 0.7,
+        dap >= 0.7 && < 0.8,
 
         haskeline >= 0.8 && < 1,
         optparse-applicative >= 0.18 && < 0.20,
@@ -235,7 +235,7 @@ executable hdb
         network-run >= 0.4.4,
         async >= 2.2.5 && < 2.3,
         text >= 2.1 && < 2.3,
-        dap >= 0.6 && < 0.7,
+        dap,
 
         haskeline >= 0.8 && < 1,
         optparse-applicative >= 0.18 && < 0.20,

